tests/handformer.test.cpp

```diff
@@ -72,3 +74,50 @@ TEST(HandFormer, HandFromNotationInvalidPieceSet) {
   auto result = HandFromNotation("799g");
   EXPECT_EQ(result, hand);
 }
+
+TEST(HandFormer, IsValidNotationWithValidNotations) {
+  EXPECT_TRUE(IsValidNotation("123s"));
+  EXPECT_TRUE(IsValidNotation("123s456m"));
+  EXPECT_TRUE(IsValidNotation("11z22z33z"));
+  EXPECT_TRUE(IsValidNotation("1p2p3p4p"));
+  EXPECT_TRUE(IsValidNotation("50m"));
+
+  // "Nothing" defaults to true
+  EXPECT_TRUE(IsValidNotation(""));
+}
+
+TEST(HandFormer, IsValidNotationWithInvaildNotations) {
+  EXPECT_FALSE(IsValidNotation("123"));
+  EXPECT_FALSE(IsValidNotation("123g456m"));
+  EXPECT_FALSE(IsValidNotation("11z22z33"));
+  EXPECT_FALSE(IsValidNotation("@#-s2712s"));
+  EXPECT_FALSE(IsValidNotation("8z"));
+}
+
+TEST(HandFormer, HandToNotationSingleSet) {
+  std::vector<Piece> hand = {Piece(Piece::ONE_BAMBOO), Piece(Piece::TWO_BAMBOO), Piece(Piece::THREE_BAMBOO)};
+  EXPECT_EQ(HandToNotation(hand), "123s");
+}
+
+TEST(HandFormer, HandToNotationMultipleSets) {
+  std::vector<Piece> hand = {
+    Piece(Piece::ONE_BAMBOO), Piece(Piece::TWO_BAMBOO), Piece(Piece::THREE_BAMBOO),
+    Piece(Piece::FOUR_PIN), Piece(Piece::FIVE_PIN), Piece(Piece::SIX_PIN),
+    Piece(Piece::NINE_CHARACTER), Piece(Piece::EAST_WIND), Piece(Piece::WEST_WIND),
+  };
+  EXPECT_EQ(HandToNotation(hand), "13z123s456p9m");
+}
+
+TEST(HandFormer, HandToNotationSkipsErrorPieces) {
+  std::vector<Piece> hand = {
+    Piece(Piece::ONE_BAMBOO), Piece(Piece::ERROR), Piece(Piece::THREE_BAMBOO),
+  };
+  EXPECT_EQ(HandToNotation(hand), "13s");
+}
+
+TEST(HandFormer, HandToNotationRedFives) {
+  std::vector<Piece> hand = {
+    Piece(Piece::FIVE_BAMBOO), Piece(Piece::RED_FIVE_BAMBOO),
+  };
+  EXPECT_EQ(HandToNotation(hand), "50s");
+}
```

Comment:
nit: /n

Reply:
fixed in 5ee5d6df3ae88121f47d7fc03d03b9139bcb821f